eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/AbstractMutableBiMap.java

```diff
@@ -153,6 +153,17 @@ else if (other == value || value.equals(other))
         return this;
     }
 
+    @Override
+    public MutableBiMap<K, V> withMap(Map<K, V> map)
+    {
+        if (map == null)
+        {
+            throw new IllegalArgumentException("Cannot Add a null Map!");
+        }
+        map.forEach(this::put);
+        return this;
+    }
```

Comment:
I don't think we should have a null check.

Instead of delegating to forEach, I think we should be delegating to putAll() which sometimes is better optimized (See UnifiedMap for an example).

Taken together, the implementation for all mutable maps will be:
```java
this.putAll(map);
return this;
```
Which means the implementations can be pulled up to interfaces as default methods. AbstractMutable*Maps will be unchanged.

Reply:
The null check is following the discussion we had previously in this PR! is just to trigger IAE instead of NPE! If the default behaviour should be to trigger NPEs instead, then I remove it and update the tests! WDYT ?

Reply:
Yes, I replied. Repeating here:

> It's not a big deal either way, but Effective Java Item 72: Favor the use of standard exceptions says "If a caller passes null in some parameter for which null values are prohibited, convention dictates that NullPointerException be thrown rather than IllegalArgumentException."

In addition, this method feels very similar to putAll(). If you call Map.putAll(null) on any map, including our own implementations, you'll get a NullPointerException. None of them have a null check either.

Reply:
Got it! Thanks

Reply:
@motlin I'm facing some test failures after removing the null check from the method! some classes (`org.eclipse.collections.impl.map.mutable.ConcurrentHashMap` for example) are triggering IAE since they use `java.util.Map#putAll` method, however others (`org.eclipse.collections.impl.bimap.mutable.HashBiMap`) trigger NPE, since `putAll` method is overridden in `org.eclipse.collections.impl.bimap.mutable.AbstractMutableBiMap`